vortex-serde/src/layouts/read/builder.rs

```diff
@@ -87,22 +88,28 @@ impl<R: VortexReadAt> LayoutBatchStreamBuilder<R> {
 
         let filter_reader = self
             .row_filter
-            .as_ref()
-            .map(|filter| {
+            .map(|row_filter| {
                 footer.layout(
-                    Scan::new(Some(Arc::new(filter.clone()))),
+                    Scan::new(Some(Arc::new(row_filter))),
                     RelativeLayoutCache::new(message_cache.clone(), footer_dtype),
                 )
             })
             .transpose()?;
 
+        let indices_mask = self
```

Comment:
you could build the mask in the `with` method and then we can allow boolean and integer masks?

Reply:
Sure, I'll do that